Formula/sftpgo.rb

```diff
@@ -0,0 +1,101 @@
+class Sftpgo < Formula
+  desc "Fully featured SFTP server with optional HTTP/S, FTP/S and WebDAV support"
+  homepage "https://github.com/drakkan/sftpgo"
+  url "https://github.com/drakkan/sftpgo/releases/download/v2.2.3/sftpgo_v2.2.3_src_with_deps.tar.xz"
+  sha256 "6c8676725e86ee3f6ad46a340a84f0da37cab8b6ea7b6aee86b2b96ba5e6671a"
+  license "AGPL-3.0-only"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-ldflags", "-s -w", "-trimpath", "-o", "sftpgo"
+    inreplace "sftpgo.json" do |s|
+      s.gsub! "\"users_base_dir\": \"\"", "\"users_base_dir\": \"#{var}/sftpgo/data\""
+      s.gsub! "\"backups_path\": \"backups\"", "\"backups_path\": \"#{var}/sftpgo/backups\""
+      s.gsub! "\"credentials_path\": \"credentials\"", "\"credentials_path\": \"#{var}/sftpgo/credentials\""
+    end
+    bin.install "sftpgo"
+    pkgetc.install "sftpgo.json"
+    pkgshare.install "static", "templates", "openapi"
+    (var/"sftpgo").mkpath
+  end
+
+  def caveats
+    <<~EOS
+      If this is your first installation please open the web administration panel:
+
+      http://localhost:8080/web/admin
+
+      and complete the initial setup.
+
+      The SFTP service is available on port 2022.
+      If the SFTPGo service does not start, make sure that TCP ports 2022 and 8080 are not used by other services
+      or change the SFTPGo configuration to suit your needs.
+
+      SFTPGo service runs as user "#{ENV["USER"]}".
+
+      Default data location:
+
+      #{var}/sftpgo
+
+      Configuration file location:
+
+      #{pkgetc}/sftpgo.json
+
+      Getting started guide:
+
+      https://github.com/drakkan/sftpgo/blob/main/docs/howto/getting-started.md
+
+      Step-to-step tutorials:
+
+      https://github.com/drakkan/sftpgo/tree/main/docs/howto
+    EOS
+  end
+
+  plist_options startup: true
+
+  def plist
```

Comment:
Can we use a `service` block instead?

Reply:
I'm not a macOS expert, so this question could be trivial. I would like to run the service as non-root user, so in the service definition I have set

```
<key>UserName</key>
<string>#{ENV["USER"]}</string>
```

how can I do this in a service block? Thanks

Reply:
That doesn't actually work in `brew services`, but `brew services` runs as non-root user by default anyway.

Reply:
strange, maybe there is something wrong in my local installation then

```
sudo brew services info sftpgo
sftpgo (homebrew.mxcl.sftpgo)
Running: ✔
Loaded: ✔
Schedulable: ✘
User: root
PID: 52423
```

this is the output I get after installing the just pushed version. As said before I don't know anything about macOS, this is the first time I use homebrew :smile: